### PR TITLE
Service tester crud

### PR DESCRIPTION
**Pull request by jaylees14**
Test the `CREATE` endpoint of each service by mocking a request body, then validating the response is what we expect:

Again running with `spec-golang` and `tinder.temple`
```
🍪 Spinning up Kubernetes infrastructure...
🧪 Testing Auth service
    ✅ Auth register
    ✅ Auth login
🧪 Testing User service
    ✅ User create
🧪 Testing Match service
    ✅ Match create
💀 Shutting down Kubernetes infrastructure...
```

### CODE REVIEW

File: src/main/scala/temple/test/internal/EndpointTest.scala [hunk 1]
```diff
@@ -1,10 +1,98 @@
...
+        max.foreach { maxValue =>
+          assert(floatValue <= maxValue, s"$key does not satisfy max bound of $maxValue")
+        }
+    }
+
+  def validateResponseBody(request: JsonObject, response: JsonObject, attributes: Map[String, Attribute]): Unit = {
+    // The response should contain exactly the keys in attributes, PLUS an ID attribute, MINUS anything that is @server
+    val expectedAttributes = attributes.filter {
+      case (_, attribute) =>
+        attribute.accessAnnotation.fold(true) {
```
Comment (c-harding) [inline src/main/scala/temple/test/internal/EndpointTest.scala]: One of these would surely be preferable to explicitly listing every option:

accessAnnotation != Some(Annotation.Server)
!accessAnnotation.contains(Annotation.Server)

File: src/main/scala/temple/test/internal/EndpointTest.scala [hunk 2]
```diff
@@ -1,10 +1,98 @@
...
+          case Annotation.Server                        => false
+          case Annotation.ServerSet | Annotation.Client => true
+        }
+    }
+    val expectedAttributeNames = (expectedAttributes.keys ++ Seq("ID")).map(_.capitalize).toSet
```
Comment (c-harding) [inline src/main/scala/temple/test/internal/EndpointTest.scala]: Flip the args of the ++, replace Seq with Set, and there’s no need for the toSet.

File: src/main/scala/temple/test/internal/EndpointTest.scala [hunk 3]
```diff
@@ -1,10 +1,98 @@
...
+    assertEqual(expectedAttributeNames, foundAttributeNames)
+
+    expectedAttributes.foreach {
+      case (name, attribute) =>
+        // Validate that the response JSON is of the expected types
+        val responseForKey = response(name.capitalize).getOrElse(fail(s"could not find key $name in response body"))
+        validateResponseType(name, responseForKey, attribute)
+
+        // Where the attribute is not @serverSet, validate that what was sent is exactly what is returned
+        val shouldValidateRequest = attribute.accessAnnotation.fold(true) {
```
Comment (c-harding) [inline src/main/scala/temple/test/internal/EndpointTest.scala]: Ditto

**Review by c-harding — APPROVED** [review]

File: src/main/scala/temple/test/internal/EndpointTest.scala (hunk 1 above)
Comment (jaylees14) [reply src/main/scala/temple/test/internal/EndpointTest.scala]: Good shout

**Review by Lewiky — APPROVED** [review]
Little hard to follow in places, but overall looks good
